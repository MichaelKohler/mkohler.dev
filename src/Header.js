import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => {
  return (
    <header>
      <section id="top-navigation" className="container-fluid nopadding">
        <div className="row nopadding ident ui-bg-color01">
          <div className="col-md-5 col-lg-4 vc-photo">&nbsp;</div>

          <div className="col-md-7 col-lg-8 vc-name nopadding">
            <div className="row nopadding name">
              <div className="col-md-12 name-title">
                <h1 className="font-accident-two-light uppercase">Michael Kohler</h1>
              </div>
            </div>

            <aside className="row nopadding position">
              <div className="col-md-12 position-title">
                <section className="cd-intro">
                  <h4 className="cd-headline clip is-full-width font-accident-two-normal uppercase">
                    <span>I&apos;m an experienced&nbsp; ...</span>
                  </h4>
                </section>
              </div>
            </aside>

            <nav>
              <ul id="nav" className="row nopadding cd-side-navigation">
                <li className="col-xs-4 col-sm-2 nopadding menuitem ui-menu-color01">
                  <Link to="/">
                    <i className="fa fa-user" />
                    <span>resume</span>
                  </Link>
                </li>
                <li className="col-xs-4 col-sm-2 nopadding menuitem ui-menu-color03">
                  <Link to="/mozilla">
                    <i className="fa fa-globe" />
                    <span>mozilla</span>
                  </Link>
                </li>
                <li className="col-xs-4 col-sm-2 nopadding menuitem ui-menu-color04">
                  <Link to="/talks">
                    <i className="fa fa-microphone" />
                    <span>talks</span>
                  </Link>
                </li>
                <li className="col-xs-4 col-sm-2 nopadding menuitem ui-menu-color05">
                  <Link to="/contact">
                    <i className="fa fa-envelope" />
                    <span>contact</span>
                  </Link>
                </li>
                <li className="col-xs-4 col-sm-2 nopadding menuitem ui-menu-color05">
                  <a
                    target="_blank"
                    rel="noopener noreferrer"
                    href="https://www.github.com/MichaelKohler"
                  >
                    <i className="fa fa-github" />
                    <span>GitHub</span>
                  </a>
                </li>
                <li className="col-xs-4 col-sm-2 nopadding menuitem ui-menu-color06">
                  <a
                    target="_blank"
                    rel="noopener noreferrer"
                    href="https://www.michaelkohler.info"
                  >
                    <i className="fa fa-pencil" />
                    <span>blog</span>
                  </a>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </section>
    </header>
  );
};

export default Header;
