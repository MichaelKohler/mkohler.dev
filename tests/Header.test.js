import React from 'react';
import { shallow } from 'enzyme';

import Header from '../src/Header';

test('should correctly render Header', () => {
  const component = shallow(<Header />);
  expect(component).toMatchSnapshot();
});
