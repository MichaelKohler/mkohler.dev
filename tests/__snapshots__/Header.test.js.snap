// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Header 1`] = `
<header>
  <section
    className="container-fluid nopadding"
    id="top-navigation"
  >
    <div
      className="row nopadding ident ui-bg-color01"
    >
      <div
        className="col-md-5 col-lg-2 vc-photo"
      >
         
      </div>
      <div
        className="col-md-7 col-lg-10 vc-name nopadding"
      >
        <div
          className="row nopadding name"
        >
          <div
            className="col-md-12 name-title"
          >
            <h1
              className="font-accident-two-light uppercase"
            >
              Michael Kohler
            </h1>
          </div>
        </div>
        <aside
          className="row nopadding position"
        >
          <div
            className="col-md-12 position-title"
          >
            <section
              className="cd-intro"
            >
              <h2
                className="cd-headline clip is-full-width font-accident-two-normal uppercase"
              >
                <span>
                  I'm an experienced 
                  <Typist
                    avgTypingDelay={70}
                    className=""
                    cursor={
                      Object {
                        "hideWhenDone": true,
                      }
                    }
                    delayGenerator={[Function]}
                    onCharacterTyped={[Function]}
                    onLineTyped={[Function]}
                    onTypingDone={[Function]}
                    startDelay={0}
                    stdTypingDelay={25}
                  >
                    <span>
                      JavaScript Developer
                    </span>
                    <Backspace
                      count={20}
                      delay={1000}
                    />
                    <span>
                      Node.js Developer
                    </span>
                    <Backspace
                      count={17}
                      delay={1000}
                    />
                    <span>
                      Community Builder
                    </span>
                    <Backspace
                      count={17}
                      delay={1000}
                    />
                    <span>
                      JavaScript Developer
                    </span>
                  </Typist>
                </span>
              </h2>
            </section>
          </div>
        </aside>
        <nav>
          <ul
            className="row nopadding cd-side-navigation"
            id="nav"
          >
            <li
              className="col-xs-4 col-sm-2 nopadding menuitem ui-menu-color01"
            >
              <Link
                to="/"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        448,
                        512,
                        Array [],
                        "f007",
                        "M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z",
                      ],
                      "iconName": "user",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                <span>
                  resume
                </span>
              </Link>
            </li>
            <li
              className="col-xs-4 col-sm-2 nopadding menuitem ui-menu-color03"
            >
              <Link
                to="/mozilla"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        496,
                        512,
                        Array [],
                        "f0ac",
                        "M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z",
                      ],
                      "iconName": "globe",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                <span>
                  mozilla
                </span>
              </Link>
            </li>
            <li
              className="col-xs-4 col-sm-2 nopadding menuitem ui-menu-color04"
            >
              <Link
                to="/talks"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        352,
                        512,
                        Array [],
                        "f130",
                        "M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z",
                      ],
                      "iconName": "microphone",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                <span>
                  talks
                </span>
              </Link>
            </li>
            <li
              className="col-xs-4 col-sm-2 nopadding menuitem ui-menu-color05"
            >
              <Link
                to="/contact"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f0e0",
                        "M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z",
                      ],
                      "iconName": "envelope",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                <span>
                  contact
                </span>
              </Link>
            </li>
            <li
              className="col-xs-4 col-sm-2 nopadding menuitem ui-menu-color05"
            >
              <a
                href="https://www.github.com/MichaelKohler"
                rel="noopener noreferrer"
                target="_blank"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        496,
                        512,
                        Array [],
                        "f09b",
                        "M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z",
                      ],
                      "iconName": "github",
                      "prefix": "fab",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                <span>
                  GitHub
                </span>
              </a>
            </li>
            <li
              className="col-xs-4 col-sm-2 nopadding menuitem ui-menu-color06"
            >
              <a
                href="https://www.michaelkohler.info"
                rel="noopener noreferrer"
                target="_blank"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f303",
                        "M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z",
                      ],
                      "iconName": "pencil-alt",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                <span>
                  blog
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </section>
</header>
`;
